server/src/test/java/org/spine3/server/stand/StandFunnelShould.java

```diff
@@ -94,10 +166,150 @@ public void fail_to_initialize_from_empty_builder() {
      */
 
+    @Test
+    public void deliver_updates_from_projection_repository() {
+        deliverUpdates(false, projectionRepositoryDispatch());
+    }
+
+    @Test
+    public void deliver_updates_form_aggregate_repository() {
+        deliverUpdates(false, aggregateRepositoryDispatch());
+    }
+
+    @Test
+    public void deliver_updates_from_several_repositories_in_single_thread() {
+        deliverUpdates(false, getSeveralRepositoryDispatchCalls());
+    }
+
+    @Test
+    public void deliver_updates_from_several_repositories_in_multiple_threads() {
+        deliverUpdates(true, getSeveralRepositoryDispatchCalls());
+    }
+
+    private static BoundedContextAction[] getSeveralRepositoryDispatchCalls() {
+        final BoundedContextAction[] result = new BoundedContextAction[Given.SEVERAL];
+        final Random random = new SecureRandom();
+
+        for (int i = 0; i < result.length; i++) {
+            result[i] = random.nextBoolean() ? aggregateRepositoryDispatch() : projectionRepositoryDispatch();
+        }
+
+        return result;
+    }
+
+    private static void deliverUpdates(boolean isMultiThread, BoundedContextAction... dispatchActions) {
+        checkNotNull(dispatchActions);
+
+        final Stand stand = mock(Stand.class);
+        final BoundedContext boundedContext = spy(Given.boundedContext(stand,
+                                                                       isMultiThread ?
+                                                                       Given.THREADS_COUNT_IN_POOL_EXECUTOR : 0));
+
+        for (BoundedContextAction dispatchAction : dispatchActions) {
+            dispatchAction.perform(boundedContext);
+        }
+
+        // Was called as much times as there are dispatch actions.
+        verify(boundedContext, times(dispatchActions.length)).getStandFunnel();
+
+        if (isMultiThread) {
+            await(Given.AWAIT_SECONDS);
+        }
+
+        verify(stand, times(dispatchActions.length)).update(ArgumentMatchers.any(), any(Any.class));
+    }
+
+    private static void await(int seconds) {
+        try {
+            Thread.sleep(seconds);
+        } catch (InterruptedException ignore) {
+        }
+    }
+
+    private static BoundedContextAction aggregateRepositoryDispatch() {
+        return new BoundedContextAction() {
+            @Override
+            public void perform(BoundedContext context) {
+                // Init repository
+                final AggregateRepository<?, ?> repository = Given.aggregateRepo(context);
+
+                repository.initStorage(InMemoryStorageFactory.getInstance());
+
+                try {
+                    repository.dispatch(Given.validCommand());
+                } catch (IllegalStateException e) {
+                    // Handle null event dispatch after command handling.
+                    Assert.assertTrue(e.getMessage().contains("No record found for command ID: EMPTY"));
```

Comment:
Why do we check that? That is not related to StandFunnel functionality.

Also, it does not read: you dispatch a `Given.validCommand()`; in this case you should have returned some event, not raise an exception.